tools/oeedger8r/Emitter.ml

```diff
@@ -327,6 +326,27 @@ let get_cast_to_mem_expr (ptype, decl)=
     else
       sprintf "(%s) " (get_tystr t)
 
+(** [get_parameter_type_str] takes a parameter and returns its type as
+    a string suitable for a C cast. It handles arrays. For example:
+    {[
+      int
+      double*
+      char foo[]
+    ]}
+    become
+    {[
+      int
+      double*
+      char*
+    ]}. NOTE: This does not yet handle multi-dimensional arrays correctly. *)
+let get_parameter_type_str (pt: Ast.parameter_type) (declr: Ast.declarator) =
+  match pt with
+  | Ast.PTPtr (atype, ptr_attr) ->
+    let tystr = Ast.get_tystr atype in
+    (* TODO: We probably need to handle multi-dimensional arrays here. *)
+    tystr ^ if Ast.is_array declr || ptr_attr.pa_isary then "*" else ""
```

Comment:
This is what broke everything else after I'd initially fixed my single repro yesterday 😆 Had to handle both normal arrays and "foreign" arrays.

Reply:
Given that now the type is being used in the LHS, how do const arguments work?

Reply:
Do we assert in our tests that the foreign arrays values are correctly marshalled?

Reply:
To the last question, it sure looks like it to me:

Reply:
To the first question, this function will never emit a `const` qualifier as `get_tystr` does not do that, and we don't explicitly check `pa_rdonly` to add it.

We could, if we needed to. Checking if that's a possibility...

Reply:
I had to dig into this quite a bit, should be good to go now.